6/6.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func readFile(name string) ([]string, error) {
+	content, err := os.ReadFile(name)
+	if err != nil {
+		log.Fatalf("Nelze načíst soubor: %v", err)
+		return nil, err
+	}
+
+	groups := strings.Split(string(content), "\n\n")
+	return groups, nil
+}
+
+func removeDuplicates(s string) string {
+	seen := make(map[rune]bool)
+	result := []rune{}
+
+	for _, char := range s {
+		if !seen[char] && char != '\n' {
+			seen[char] = true
```

Comment:
Návrh na optimalizaci – pokud potřebuješ jen uložit key (v tvém případě `rune`), který se např. vyskytl nebo který _něco_, je efektivnější metoda – namísto `bool` stačí použít prázdnou stukturu `struct{}`. Prázdná struktura zabíra v paměti 0 bitů, zatímco boolean hodnota zabírá 1 bajt. U takto malého projektu to nevadí, ale pokud bys takto ukládal velké množství hodnot, je použití `struct{}` lepší.

Takto by vypadal upravený kód:
```go
func removeDuplicates(s string) string {
	seen := make(map[rune]struct{})
	result := []rune{}
	for _, char := range s {
		if _, exists := seen[char]; !exists && char != '\n' {
			seen[char] = true
			result = append(result, char)
		}
	}
	return string(result)
}
```

Druhá možnost je použít přímo `rune` slice (`result`):

```go
func removeDuplicates(s string) string {
	result := []rune{}
	for _, char := range s {
		if !slices.Contains(result, char) && char != '\n' {
			result = append(result, char)
		}
	}
	return string(result)
}
```